### PR TITLE
cmd/devp2p: fix flakey tests in CI

### PR DESCRIPTION
**Pull request by renaynay**
This PR fixes a couple of issues in the eth test suite that caused flakiness when run in the CI. 

It depends on #22754 getting merged.

### CODE REVIEW

File: cmd/devp2p/internal/ethtest/suite_test.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func TestEthSuite(t *testing.T) {
 				t.Fatal()
 			}
 		})
+		time.Sleep(100 * time.Millisecond)
```
Comment (holiman) [inline cmd/devp2p/internal/ethtest/suite_test.go:55]: This is a bit indicative of it still being flaky. I'm guessing you added it to make the spurious packets from earlier tests to interfere with later tests, but is it still needed? 
If it's only once in a while, maybe we can either figure out what spurious packets those are, and just 'continue' on them, _or_ maybe somehow drain the deliver buffer between test executions? 

Just having a random 100ms timout like that means it'll work "more often", but still spuriously fail on e.g. appveyor. 

Not a blocker, but I would prefer if we solved it without having that in

**Review by holiman — APPROVED** [review]
Generally I'm ok with it as it is, if it solves the problems, but would prefer if we could get rid of the random timeout

File: cmd/devp2p/internal/ethtest/suite_test.go (hunk 1 above)
Comment (renaynay) [reply cmd/devp2p/internal/ethtest/suite_test.go:55]: I agree.

**Comment by holiman** [comment]
LGTM!

File: cmd/devp2p/internal/ethtest/types.go [hunk 2]
```diff
@@ -334,17 +334,18 @@ loop:
...
 		req := &GetBlockHeaders{Origin: eth.HashOrNumber{Hash: block.Hash()}, Amount: 1}
 		if err := c.Write(req); err != nil {
 			return err
 		}
 		switch msg := c.Read().(type) {
 		case *BlockHeaders:
-			if len(*msg) > 0 {
-				return nil
+			for _, header := range *msg {
+				if header.Number.Uint64() == block.Header().Number.Uint64() {
```
Comment (holiman) [inline cmd/devp2p/internal/ethtest/types.go:346]:
```suggestion
				if header.Number.Uint64() == block.NumberU64() {
```
I think we have a convenience function for that^

**Comment by holiman** [comment]
Unfortunately, still flaky on Appveyor: 
```
not ok 1 TestBroadcast_66
# wrong head block in status, want:  0x8c795a2497f393359fd66bfd5696442d12a81ccb1110dffd636604bfc9af4df3 (block 1000) have 0x0e70e01064023f70f047dbf0b97e7109e4aa5df3d643d45f8e91e47d3d67a424
```

File: cmd/devp2p/internal/ethtest/eth66_suiteHelpers.go [hunk 3]
```diff
@@ -319,10 +319,10 @@ func (s *Suite) sendNextBlock66(t *utesting.T) {
 	}
 	// send announcement and wait for node to request the header
 	s.testAnnounce66(t, sendConn, receiveConn, blockAnnouncement)
-	// update test suite chain
-	s.chain.blocks = append(s.chain.blocks, s.fullChain.blocks[nextBlock])
 	// wait for client to update its chain
 	if err := receiveConn.waitForBlock66(s.chain.Head()); err != nil {
```
Comment (holiman) [inline cmd/devp2p/internal/ethtest/eth66_suiteHelpers.go:323]: Shouldn't this be ?
```suggestion
	if err := receiveConn.waitForBlock66(s.fullchain.blocks[nextBlock]); err != nil {
```
I really don't know, just a thought..?

**Comment by holiman** [comment]
i think you also need to apply this: https://github.com/renaynay/go-ethereum/blob/b5eb4d1195849a8ca75fbdeb5976fe7234d0e142/cmd/devp2p/internal/ethtest/types.go#L345 here: https://github.com/ethereum/go-ethereum/blob/ef67673b1ee9d7597f6dcb7b19c65c2c32618dc9/cmd/devp2p/internal/ethtest/eth66_suiteHelpers.go#L256

**Comment by holiman** [comment]
Ah, so the error is in the status exchange, where the 'other side' has a lower block  than we expected. Isn't it because the previous broacast test already progressed 'us' one block?

**Pull request merged**